### PR TITLE
[Enhancement] Implement remove_first_n_values of MapColumn and ArrayColumn

### PR DESCRIPTION
**Pull request by liuyehcf**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Seaven — DISMISSED** [review]

**Review by murphyatwork — DISMISSED** [review]

File: be/src/column/array_column.cpp [hunk 1]
```diff
@@ -213,6 +213,16 @@ Status ArrayColumn::update_rows(const Column& src, const uint32_t* indexes) {
     return Status::OK();
 }
 
+void ArrayColumn::remove_first_n_values(size_t count) {
+    size_t offset = _offsets->get_data()[count];
```
Comment (fzhedu) [inline be/src/column/array_column.cpp:217]: what if count > size()?

Comment (liuyehcf) [reply be/src/column/array_column.cpp]: Updated

**Review by silverbullet233 — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by fzhedu — APPROVED** [review]
LGTM

**Review by silverbullet233 — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**